reference_system_autoware/CMakeLists.txt

```diff
@@ -30,23 +30,35 @@ ament_auto_add_executable(autoware_default_multithreaded
 )
 
 if(${BUILD_TESTING})
-  # run test psrecord to generate KPIs
-  # arguments for script are as follows:
-  # 1. executable name
-  # 2. path to save log data
-  # 3. path to save plot
-  # 4. duration to run test
-  #  add_custom_command(TARGET autoware_default_singlethreaded
-  #    POST_BUILD
-  #    COMMAND
-  #      ${CMAKE_CURRENT_SOURCE_DIR}/test/generate_graph.sh
-  #      autoware_default_singlethreaded
-  #      ${CMAKE_CURRENT_SOURCE_DIR}/results/cpu_mem_usage_results.txt
-  #      ${CMAKE_CURRENT_SOURCE_DIR}/results/plot.png
-  #      5
-  #    ARG
-  #    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  #    COMMENT "Run psrecord for specified target")
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  # run target for 5 seconds
+  function(test_reference_system_autoware_5s target)
```

Comment:
this function should allow us to pass in different executables and run the same unit tests on them i think